### PR TITLE
Add `RedisConversationMemoryDriver`

### LINKED ISSUE #389: Redis Conversation Memory Driver
Users want to store Conversation Memory in other databases besides DynamoDb. Redis is a fast, flexible, and popular option. I propose adding a `RedisConversationMemoryDriver`.

### PR DESCRIPTION
**Pull request by vachillo**
closes #710 
closes #389 

----
📚 Documentation preview 📚: https://griptape--787.org.readthedocs.build//787/

### CODE REVIEW

File: docs/griptape-framework/drivers/conversation-memory-drivers.md [hunk 1]
```diff
@@ -45,3 +45,33 @@ agent = Agent(conversation_memory=ConversationMemory(driver=dynamodb_driver))
...
+from griptape.memory.structure import ConversationMemory
+from griptape.structures import Agent
+
+conversation_id = uuid.uuid4().hex
+redis_conversation_driver = RedisConversationMemoryDriver(
+    host="127.0.0.1",
+    port=6379,
+    password='',
+    index='griptape_converstaion',
+    conversation_id = conversation_id
```
Comment (collindutter) [inline docs/griptape-framework/drivers/conversation-memory-drivers.md:70]: Can you please move these to env vars and update github integ tests file?

File: pyproject.toml [hunk 2]
```diff
@@ -93,6 +94,7 @@ drivers-web-scraper-markdownify = ["playwright", "beautifulsoup4", "markdownify"
 drivers-event-listener-amazon-sqs = ["boto3"]
 drivers-event-listener-amazon-iot = ["boto3"]
 
+
```
Comment (collindutter) [inline pyproject.toml:97]: Nit: extra newline.

File: poetry.lock
Comment (collindutter) [inline poetry.lock:1]: This seems like an overly-changed lock file for just adding redis. Can you try regenerating using the one from `dev`?

Comment (vachillo) [reply poetry.lock:1]: copied from dev and ran `poetry lock --no-update`

**Review by collindutter — APPROVED** [review]

**Pull request merged**